### DISCUSSION TITLE
Shadowsocks UoT Version 1 or 2 ？

### DISCUSSION THREAD

**Discussion by MFWT** [XTLS/Xray-core#2528] [Q&A] [answered]
Xray对Shadowsocks提供了UoT（UDP over TCP）的支持，这很好，毕竟就国内目前对UDP的流量限制而言是有帮助的

但是查阅官方文档可以发现，UoT支持两个版本，1和2，但是文档中并没有写出1和2的区别，以及兼容性如何

我的问题是，**在客户端与服务端均保证是最新的且相同的Xray版本的情况下**，应该选哪个UoT版本比较好？以及他们大概是有什么差别，比如性能，NAT行为等？

望不吝赐教

**Comment by chika0801** [comment, marked as answer]
> 以及他们大概是有什么差别，比如性能，NAT行为等？

这个是 sing-box 作者提交的PR，建议你去那边的github或tg群问问。

> 在客户端与服务端均保证是最新的且相同的Xray版本的情况下，应该选哪个UoT版本比较好？

保证两端一致，即非一边V1一边V2就行了吧。
